Sources/NIOSingletonsPosix/PosixSingletons.swift

```diff
@@ -0,0 +1,66 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_exported import NIOSingletonResources
+import NIOPosix
+import NIOCore
+
+extension NIOSingletons {
+    /// A globally shared, lazily initialized ``MultiThreadedEventLoopGroup``  that uses `epoll` as the selector mechanism.
+    ///
+    /// The number of threads is determined by ``NIOSingletons.suggestedMultiThreadedEventLoopGroupThreadCount``.
+    public static var multiThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup {
+        return globalMultiThreadedPosixEventLoopGroup
+    }
+
+    /// A globally shared, lazily initialized ``MultiThreadedEventLoopGroup`` with just one thread that uses `epoll` as the selector mechanism.
+    public static var singleThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup {
+        return globalSingleThreadedPosixEventLoopGroup
+    }
+
+    /// A globally shared, lazily initialized ``EventLoop`` that uses `epoll` as the selector mechanism. Backed by `NIOSingletons.singleThreadedPosixEventLoopGroup`.
+    public static var posixEventLoop: EventLoop {
+        return Self.singleThreadedPosixEventLoopGroup.next()
+    }
+
+    /// A globally shared, lazily initialized ``NIOThreadPool`` that can be used for blocking I/O and other blocking operations.
+    ///
+    ///     /// The number of threads is determined by ``NIOSingletons.suggestedBlockingPoolThreadCount``.
+    public static var posixBlockingPool: NIOThreadPool {
```

Comment:
`posixBlockingPool` doesn't mean much to me. Can we name this something different, `posixBlockingThreadPool` maybe? I'm guessing `posixThreadPool` is out of the question

Reply:
@adam-fowler The default way is now `NIOThreadPool.globalSingleton` which I suppose is clearer?